Added Go project and instructions [API-404]

yuce opened the pull request:
~`go.mod` contains a tag I've created in my personal repo. I'll updated it when we release the Go client preview 1 tomorrow.~
`go.mod` is up to date with the latest Preview 1

yuce commented:
Updated the PR with the v1 preview 1 release.

leszko reviewed (changes requested):
Thanks for the PR @yuce . I feel like we almost have all clients covered!!! 🥇 

Added one inline comment, other than that, 2 things:
1. Could you add `go` to integration tests [here](https://github.com/hazelcast-guides/kubernetes/blob/master/.github/workflows/integration-tests.yml#L17)
2. I checked some scenarios and it seems that the go client prints some weird logs after a Hazelcast cluster is restarted. You can find details [here](https://hazelcast.atlassian.net/wiki/spaces/PM/pages/2849079475/Hazelcast+Clients+in+Kubernetes+TDD) (section "Go Client", at the bottom). Could you please check it?

yuce commented on doc/modules/ROOT/pages/index.adoc:
```diff
@@ -206,6 +215,28 @@ auto hz = hazelcast::new_client(std::move(config)).get();
+
+func main() {
+    cb := hazelcast.NewConfigBuilder()
+    cb.Cluster().SetAddrs("hz-hazelcast:5701")
```
Port is required in v1.0.0, but it will be possible in v1.1.0 to leave out the port or specify a port range.

yuce commented:
@leszko  Thanks a lot for the review! I've added Go to integration tests and updated the code for v1.0.0. The weird log errors when the cluster is totally disconnected should be fixed, but when some of the members got disconnected, the client still tries to use cached addresses for those members and will print "Found a broken connection to..." to log. I've created this ticket to track that issue: https://github.com/hazelcast/hazelcast-go-client/issues/600

leszko reviewed (approved):
Thanks for all the changes and for creating the related GH Issue. LGTM 👍

yuce commented:
Thanks!
